Mathlib/SetTheory/Surreal/Division.lean

```diff
@@ -0,0 +1,714 @@
+/-
+Copyright (c) 2024 Theodore Hwa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Theodore Hwa
+-/
+import Mathlib.SetTheory.Surreal.Multiplication
+import Mathlib.Tactic.Ring.RingNF
+import Mathlib.Tactic.FieldSimp
+
+/-!
+# Surreal division
+
+In this file, we prove that if `x` is a nonzero surreal number, then `x⁻¹` (defined in
+`Mathlib.SetTheory.Game.Basic`) is a number and is a multiplicative inverse for `x`. We use that
+to define the field structure on `Surreal`.
+
+We essentially follow the proof in ONAG [Conway2001], chapter 1, section 10, with a few minor
+differences explained below. There are four lemmas stated there, (i) through (iv). They are only
+proved for positive numbers `x`. Once we have defined the inverse for positive `x`, it is extended
+in the obvious way to negative numbers.
+
+The desired final results are (ii) and (iv) which state that `x⁻¹` is a number and is a
+multiplicative inverse, respectively.
+
+In our proof, lemmas (i) through (iv) are proved for positive numbers in the section labeled
+`Positive` by a series of preliminary lemmas, culminating in the lemma `onag_1_10` which
+proves (ii) and (iv) by a joint induction over the two statements. Lemmas (i) and (iii) are used
+only within the proof of `onag_1_10` and are not needed as induction hypotheses.
+
+We do not use the notation `x⁻¹` in the proof of lemmas (i) to (iv); we use `x.inv'` instead
+since that is the definition for positive `x`. We only use `x⁻¹` at the very end when we define
+the inverse for all nonzero surreals, both positive and negative.
+
+## Differences with ONAG
+
+Conway starts by replacing `x` with an equal game which I have called its `normalization`.
+However, the normalization is not actually needed until parts (iii) and (iv) of the proof,
+so we do not use it until those parts.
+
+For convenience, we split (iii) into 2 lemmas, the Left and Right statements.
+
+The proof of (iii) in ONAG has a minor gap. It does not work for the Left option `x' = 0`. However,
+in that case, (iii) reduces to (i). It is treated as a separate case in our proof.
+
+## Notation
+
+Throughout the proof of lemmas (i) to (iv), we use the same notation as in ONAG:
+
+`x` : the positive number whose inverse we are constructing
+`x'` : a positive option of `x`, indexed by `i`
+`y` : the purported inverse of `x`, as defined by `x.inv'`
+`y'` : an option of `y`, indexed by `i'`
+`y''` : `(1 + (x' - x) * y') / x'` , the form of any option of `y` besides 0, indexed by `i''`
+
+## References
+
+* [Conway, *On numbers and games*][Conway2001]
+
+-/
+open SetTheory
+open PGame
+open Equiv
+
+variable {x : PGame} (x_num : x.Numeric)
+
+namespace Surreal.Division
+section Positive
+
+variable (x_pos : 0 < x)
```

Comment:
It's not good practice to have conditions on variables. You might then miss cases when they're not strictly necessary. For instance, `Numeric.normalization` only requires `0 ≤ x` (though I guess it does make sense to instead require `0 < x`).